### PR TITLE
Continue on the Nullcheck remove fix

### PR DESCRIPTION
**Pull request by xuzhg**
### Issues

*This pull request fixes issue #2284.*

### Description

*Briefly describe the changes of this pull request.*

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs [hunk 1]
```diff
@@ -69,11 +69,11 @@ public static DataSourceProviderKind GetDataSourceProviderKind(this IQueryable q
             string queryProviderNamespace = query.Provider.GetType().Namespace;
             switch (queryProviderNamespace)
```
Comment (gathogojr) [inline src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs:70]: @xuzhg Given that you added `DataSourceProviderKind.InMemory` should the default case still return `DataSourceProviderKind.None`

Comment (xuzhg) [reply src/Microsoft.AspNet.OData.Shared/Query/HandleNullPropagationOptionHelper.cs:70]: Emm, maybe not. InMemory is not for the "IQueryable", this method is for the IQueryable. So, it's better to return "None". @gathogojr

**Review by gathogojr — APPROVED** [review]
LGTM

**Review by mikepizzo — APPROVED** [review]
:shipit:

**Pull request merged**